sql/src/main/java/io/crate/protocols/postgres/types/VarCharType.java

```diff
@@ -30,24 +30,42 @@
 class VarCharType extends PGType {
 
     public static final PGType INSTANCE = new VarCharType();
-    static final int OID = 1043;
 
-    private static final int TYPE_LEN = -1;
+    static final int VCHAR_OID = 1043;
+    private static final int VCHAR_TYPE_LEN = -1;
     private static final int TYPE_MOD = -1;
 
+    static final int NAME_OID = 19;
+    private static final int NAME_ARRAY_OID = -1;
+    private static final int NAME_TYPE_LEN = 64;
+
+    static final PGType NAME_INSTANCE = new VarCharType(NAME_OID, NAME_TYPE_LEN, "name");
+
+    private Integer typArray = null;
+
     private VarCharType() {
-        super(OID, TYPE_LEN, TYPE_MOD, "varchar");
+        super(VCHAR_OID, VCHAR_TYPE_LEN, TYPE_MOD, "varchar");
+    }
+
+    private VarCharType(int oid, int maxLength, String aliasName) {
+        super(oid, maxLength, TYPE_MOD, aliasName);
     }
 
     @Override
     public int typArray() {
-        return PGArray.VARCHAR_ARRAY.oid();
+        if (typArray == null) {
+            typArray = (NAME_OID == oid())
```

Comment:
Not a big fan of this 2 phase initialisation (constructor and here). It would be better addressed with a subclass imho.
The issue here seems to be that there is no guarantee that PGArray.VARCHAR_ARRAY is instantiated at constructor time (both instances are static). Any other ideas?

Reply:
How about adding a new parameter to the constructor and pass it in at construction time?

I'd generally move any `name` specific information out of this class and create a instance for name outside.

Reply:
ok, done, however, I had to move away from instansiating `PGArray.VARCHAR_ARRAY` ta this point  of time, as this requires the instance of the VarCharType itself.